src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,28 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.Optional;
+import mate.academy.hibernate.relations.dao.CountryDao;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao;
+
+    public CountryServiceImpl(CountryDao countryDao) {
+        this.countryDao = countryDao;
+    }
+
     @Override
     public Country add(Country country) {
-        return null;
+        return countryDao.add(country);
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        Optional<Country> country = countryDao.get(id);
+        if (country.isEmpty()) {
+            return null;
+        }
+        return country.get();
```

Comment:
Return null from Optional makes no sense in Optional
Most common way is to throw proper exception in case of empty optional
```suggestion
        return countryDao.get(id).orElseThrow(() -> new EntityNotFoundException("some valid message"));
```